### PR TITLE
Improve ecs_ecr module to handle image_tag_mutability feature

### PR DESCRIPTION
**Pull request by maastiff**
##### SUMMARY
Since 2019 jul. 26, AWS Elastic Container Registry can be configured to be immutable that prevent overwritting of an existing tag in the registry.
https://aws.amazon.com/fr/about-aws/whats-new/2019/07/amazon-ecr-now-supports-immutable-image-tags/
https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-tag-mutability.html

By default, ecr is created MUTABLE, as this was the only option before this feature.
This module leverage new capabilities of boto3 to configure image_tag_mutability option.
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecr.html#ECR.Client.put_image_tag_mutability

The image_tag_mutability option definition was inspired by existing terraform module.
https://www.terraform.io/docs/providers/aws/r/ecr_repository.html

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ecs_ecr.py

##### ADDITIONAL INFORMATION
This feature request add a new parameter `image_tag_mutability` that could be `mutable` or `immutable`. By default, this option is considered `mutable`.

See example below:
```paste below
- name: create immutable ecr-repo
  ecs_ecr:
    name: super/cool
    image_tag_mutability: immutable
```

### CODE REVIEW

**Comment by rrey** [comment]
LGTM

**Review by jillr — APPROVED** [review]
Thanks @Maastiff, lgtm and tests pass on an unrestricted aws account. I'll merge this after I merge and deploy https://github.com/mattclay/aws-terminator/pull/59 to our CI environment along with some other policy changes I need to review later this week.

**Comment by Maastiff** [comment]
Thanks @jillr !

**Pull request merged**
